lib/matplotlib/tests/test_style.py

```diff
@@ -135,6 +152,23 @@ def test_context_with_badparam():
         assert mpl.rcParams[PARAM] == other_value
 
 
+def test_parent_stylesheets():
+    style_dir = os.path.join(os.path.dirname(__file__), 'styles')
+    with temp_style_dir(style_dir):
+        with style.context('main_test'):
+            assert mpl.rcParams['lines.linewidth'] == 5.0
+            assert mpl.rcParams['lines.marker'] == 'o'
+            assert mpl.rcParams['lines.linestyle'] == '--'
+            assert mpl.rcParams['lines.markersize'] == 12
+            assert mpl.rcParams['axes.facecolor'] == 'green'
+            assert mpl.rcParams['axes.linewidth'] == 5.0
+            assert mpl.rcParams['axes.grid'] == True
+            assert mpl.rcParams['axes.labelweight'] == 'bold'
+            assert mpl.rcParams['axes.labelcolor'] == 'yellow'
+            assert mpl.rcParams['axes.color_cycle'] == [u'r', u'g', u'b',
+                                                            u'c', u'y']
```

Comment:
Sorry, I was a bit unclear. I just meant that it's probably only necessary to test for a single parent. I could also see the argument for testing a grandparent, but I don't think it's strictly necessary.The main point though, was that separating the test data in external files makes the tests difficult to verify. Also, a single test here is actually testing a number of different cases (inheritance from a parent, inheritance from a grandparent, multiple parents, multiple grandparents). It would be nice to separate these out into individual tests.